### ISSUE TITLE
Cannot create nested empty array even when extending.

### ISSUE THREAD

**Issue by Riobe** [json-schema-faker/json-schema-faker#283]
Description
==

I have extended json-schema-faker like the following:

```javascript
const jsf = require('json-schema-faker');
jsf.extend('custom', () => {
  return {
    'empty_array': () => {
      return []; 
    }   
  };  
});

jsf.resolve({
  type: 'array',
  items: [
    {
      type: 'array',
      custom: 'empty_array'
    }
  ]
}).then(console.log);
```

Because of, I think, [this line](https://github.com/json-schema-faker/json-schema-faker/commit/41bc021919b57ba74ca863b3168578916b60cc41#diff-1d690af890dd5d530ffd694162dc4b2bR119) my array is being removed from the resulting array because of it's 0 length.

For my use case: I am trying to fake the result of an existing api that has array of values where every row represents an object in my database and some of those rows have nested arrays that may have no items. This behavior makes it impossible for me to model that.

I believe this will also hit if I define the items and have a `minItems` of 0 and `maxItems` of anything. If there happens to be 0 items then the array will be removed entirely and the indexes of the rest of the items will now be wrong. I can't control the API I'm mimicking so I can't change that the indexes are significant.

Expected
==

`[[]]`

Actual
==

`[]`

**Comment by pateketrueke** [comment]
Interesting case, I was reviewing your PR before reading this.

I think that adding an option for clean-or-not empty arrays can be useful.

**Comment by pateketrueke** [comment]
Fixed on `0.5.0-rc6`, closing.
